hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreSetup.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * Setup HiveMetastore. It does not create any database. All the tests should create a database
+ * accordingly. it should replace the existing setUp class {@link HiveMetastoreTest}
+ */
+class HiveMetastoreSetup {
```

Comment:
why is this class needed? why not just initialize everything in the respective test class that needs this?

Reply:
+1, I too gave same comment ( I didn't see your due to race condition)

Reply:
Intent here was to deprecate [HiveMetastoreTest](https://github.com/apache/iceberg/blob/main/hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreTest.java) and start using `HiveMetastoreSetup ` to the required tests. 
Right now this is going to be used for `TestHiveCatalog` and upcoming `[TestHiveViewCatalog](https://github.com/apache/iceberg/pull/8907/files#diff-3613527b53afae01af4bbe834e98a0439b4cf290616d66fc3ea9f28abc2bba03)`.

Reply:
the conclusion here is that we should remove this class. As I mentioned in another comment, the alternative would be to have a JUnit5 extension (similar to `NessieJaxRsExtension`) that would to the proper setup/teardown for tests